File: app/src/main/java/com/semanientreprise/soundrecorderbox/SavedRecordingsViewModel.java
package com.semanientreprise.soundrecorderbox;

import android.arch.lifecycle.ViewModel;

import io.objectbox.Box;
import io.objectbox.android.ObjectBoxLiveData;

public class SavedRecordingsViewModel extends ViewModel {

    private ObjectBoxLiveData<Recordings> recordingsLiveData;

    public ObjectBoxLiveData<Recordings> getRecordingsLiveData(Box<Recordings> recordingsBox) {
        if (recordingsLiveData == null) {
            recordingsLiveData = new ObjectBoxLiveData<>(recordingsBox.query().build());
        }
        return recordingsLiveData;
    }
}

File: app/src/main/java/com/semanientreprise/soundrecorderbox/SavedRecordingsViewModel.kt
package com.semanientreprise.soundrecorderbox

import android.arch.lifecycle.ViewModel
import io.objectbox.Box
import io.objectbox.android.ObjectBoxLiveData

class SavedRecordingsViewModel : ViewModel() {
    private var recordingsLiveData: ObjectBoxLiveData<Recordings?>? = null
    fun getRecordingsLiveData(recordingsBox: Box<Recordings?>): ObjectBoxLiveData<Recordings?> {
        if (recordingsLiveData == null) {
            recordingsLiveData = ObjectBoxLiveData(recordingsBox.query().build())
        }
        return recordingsLiveData
    }
}